File: backend/src/main/java/com/military/doc/modules/document/entity/DocVersion.java
package com.military.doc.modules.document.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("doc_version")
public class DocVersion {
    @TableId(type = IdType.AUTO)
    private Long id;

    private Long docFileId;
    private String versionNo;
    private String sourceType;
    private Long baseVersionId;
    private Long fileObjectId;
    private String versionStatus;
    private Integer optimisticVersion;
    private Long submitUserId;
    private LocalDateTime submitTime;
    private String changeSummary;

    @TableField(fill = FieldFill.INSERT)
    private Long createdBy;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private Long updatedBy;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    @TableLogic
    private Integer deleted;
}

File: frontend/src/api/doc-version.ts
import api from './index'

export interface DocVersionItem {
  id?: number
  docFileId: number
  versionNo: string
  sourceType: string
  baseVersionId: number
  fileObjectId: number
  versionStatus: string
  optimisticVersion: number
  submitUserId: number
  submitTime: string
  changeSummary: string
}

export function getDocVersions(docFileId: number) {
  return api.get(`/doc-versions/doc-file/${docFileId}`)
}

export function getDocVersion(id: number) {
  return api.get(`/doc-versions/${id}`)
}

export function createDocVersion(data: DocVersionItem) {
  return api.post('/doc-versions', data)
}

export function updateDocVersion(id: number, data: DocVersionItem) {
  return api.put(`/doc-versions/${id}`, data)
}
